ion-schema/src/main/kotlin/com/amazon/ionschema/model/TypeArgument.kt

```diff
@@ -0,0 +1,46 @@
+package com.amazon.ionschema.model
+
+/**
+ * A TypeArgument represents (defines or references) a Type, allowing the Type to be used as an argument for a constraint.
+ *
+ * **_Do not implement this interface_**—It will become a sealed interface once `ion-schema-kotlin` is updated to use
+ * language version 1.5.
+ */
+@ExperimentalIonSchemaModel
+sealed class TypeArgument {
```

Comment:
I like `Argument` or `Parameter` maybe for this.

Reply:
Some constraints have arguments that are not types, however, so I wanted to qualify it with the word "Type". Generally, "parameter" describes a function's signature whereas "argument" refers to the variables/values being passed to a function, and that's how I arrived at `TypeArgument`

Reply:
Yea. I should've been more clear that I agree with `Type` as the prefix.